### PR TITLE
PWN-4397  - Implement Bottom navigation

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4397

## Description of Work

Implement bottom nav bar at home screen

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/MainFragment.kt [hunk 1]
```diff
@@ -78,23 +78,21 @@ class MainFragment : BaseFragment(R.layout.fragment_main), MainTabsSwitcher {
 
     override fun navigate(itemId: Int) {
         if (!fragments.containsKey(itemId)) {
-            val fragment = when (Tabs.fromTabId(itemId)) {
-                Tabs.HOME -> {
+            val fragment = when (ScreenTab.fromTabId(itemId)) {
+
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/MainFragment.kt:82]:
```suggestion
```

File: app/src/main/java/org/p2p/wallet/home/MainFragment.kt [hunk 2]
```diff
@@ -78,23 +78,21 @@ class MainFragment : BaseFragment(R.layout.fragment_main), MainTabsSwitcher {
...
-                Tabs.SEND -> {
-                    analyticsInteractor.logScreenOpenEvent(ScreenNames.Send.MAIN)
-                    SendFragment.create()
-                }
-                Tabs.SETTINGS -> {
+                ScreenTab.SETTINGS_SCREEN -> {
                     analyticsInteractor.logScreenOpenEvent(ScreenNames.Settings.MAIN)
                     SettingsFragment.create()
                 }
+                else -> throw IllegalStateException("No tab found for $itemId")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/MainFragment.kt:95]:
```suggestion
                else -> error("No tab found for $itemId")
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitBottomNavigationView.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    val menu: Menu
+        get() = binding.bottomNavigationView.menu
+
+    fun setOnItemSelectedListener(block: (ScreenTab) -> Boolean) {
+        binding.bottomNavigationView.setOnItemSelectedListener {
+            val foundTab = ScreenTab.fromTabId(it.itemId)
+            if (foundTab != null) {
+                block.invoke(foundTab)
+            }
+            return@setOnItemSelectedListener true
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitBottomNavigationView.kt:30]:
```suggestion
            ScreenTab.fromTabId(it.itemId)?.let { block.invoke(it) }
            true
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitBottomNavigationView.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        binding.buttonCenterAction.setOnClickListener { block.invoke() }
+    }
+
+    fun inflateMenu(menuResId: Int) {
+        binding.bottomNavigationView.inflateMenu(menuResId)
+    }
+}
+
+enum class ScreenTab(val itemId: Int) {
+    HOME_SCREEN(R.id.homeItem),
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitBottomNavigationView.kt:44]: there are 4 tabs in total, but there are only 3 here, is it ok?

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitBottomNavigationView.kt (hunk 4 above)
Comment (davran1001) [reply ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitBottomNavigationView.kt:44]: Feedback tab opens intercom, that's why i removed that one

**Pull request merged**